src/app/view/directives/dot-maxlength/dot-maxlength.directive.spec.ts

```diff
@@ -20,28 +19,42 @@ describe('DotMaxlengthDirective', () => {
 
         fixture = TestBed.createComponent(TestComponent);
         element = fixture.debugElement.query(By.css('div'));
-        directiveInstance = element.injector.get(DotMaxlengthDirective);
     });
 
-    it('should trigger event handler on paste', () => {
-        spyOn(directiveInstance, 'eventHandler');
-        element.triggerEventHandler('paste', 'Longer text example');
-        expect(directiveInstance.eventHandler).toHaveBeenCalled();
-    });
-
-    it('should trigger event handler on keypress', () => {
-        spyOn(directiveInstance, 'eventHandler');
-        element.triggerEventHandler('keypress', 'Longer text example');
-        expect(directiveInstance.eventHandler).toHaveBeenCalled();
-    });
+    it(
+        'should allow new characters when max length is not reached',
+        fakeAsync(() => {
+            const event = new Event('keypress');
+            fixture.detectChanges();
+            element.nativeElement.dispatchEvent(event);
+            element.nativeElement.textContent = element.nativeElement.textContent + 'o';
+            tick(5);
+            expect(element.nativeElement.textContent).toBe('testo');
+        })
+    );
 
     it(
-        'should remove extra characters',
+        'should remove extra characters when length is more than max length',
         fakeAsync(() => {
+            const event = new Event('paste');
             fixture.detectChanges();
-            directiveInstance.eventHandler();
+            element.nativeElement.dispatchEvent(event);
+            element.nativeElement.textContent = '12345678901';
```

Comment:
And the same here... shouldn't the `textContent` be updated with the event dispatched?